lib/winapi/filemanip.c

```diff
@@ -235,3 +239,80 @@ BOOL MoveFileA (LPCTSTR lpExistingFileName, LPCTSTR lpNewFileName)
         return TRUE;
     }
 }
+
+DWORD GetLogicalDrives (VOID)
+{
+    DWORD result = 0;
+    NTSTATUS status;
+    ANSI_STRING path;
+    HANDLE handle;
+    OBJECT_ATTRIBUTES attributes;
+    struct {
+        OBJECT_DIRECTORY_INFORMATION objDirInfo;
+        CHAR filenameBuf[100];
```

Comment:
100 is a bit arbitrary? Why not keep this shorter? We check for length of 2 shortly afterwards anyway.

If there's a custom device with a longer name (is that possible?) this would break this function, too.

Instead, why not just ignore entries which are too long? I'd assume that `NtQueryDirectoryObject` would report some error if the buffer is too short?

Reply:
I had to check whether we can ignore entries when it returns `STATUS_BUFFER_TOO_SMALL`, turns out we can, so the buffer is now only 2 bytes.